lib/js/stacks.js

```diff
@@ -13,13 +13,25 @@
     });
     StacksController.prototype.getElementData = function(element, key) {
         return element.getAttribute("data-" + this.identifier + "-" + key);
-    }
+    };
     StacksController.prototype.setElementData = function(element, key, value) {
         element.setAttribute("data-" + this.identifier + "-" + key, value);
-    }
+    };
     StacksController.prototype.removeElementData = function(element, key) {
         element.removeAttribute("data-" + this.identifier + "-" + key);
-    }
+    };
+    StacksController.prototype.triggerEvent = function(eventName, optionalElement) {
```

Comment:
It's not strictly needed here, but could we upgrade this to a custom event so we can pass data?

```
    StacksController.prototype.triggerEvent = function(eventName, detail, optionalElement) {
	        var event;
	        var namespacedName = this.identifier + ":" + eventName;
	        try {
	            event = new CustomEvent(namespacedName, {bubbles: true, detail: detail});
	        } catch (ex) {
	            // Internet Explorer
	            event = document.createEvent("CustomEvent");
	            event.initCustomEvent(namespacedName, true, true, detail);
	        }
	        (optionalElement || this.element).dispatchEvent(event);
	    };
```

I have a controller that wants to post a deletion event for another to read:

```
Stacks.addController('foo', {
   removeItem(e) {
    var el = e.currentTarget;
    var id = this.getElementData(el, 'id');
    el.parentNode.removeChild(el);
    this.triggerEvent('removed', { id: id });
  },
});

Stacks.addController('bar', {
   logRemoval(e) {
    var id = e.detail.id;
    this.element.textContent = `${id} removed`;
  },
});
```

```
<div data-controller="foo">
  <button data-foo-id="123" data-action="removeItem">Remove Me</button>
</div>
<div data-controller="bar" data-action="foo:removed@document->bar#logRemoval"></div>
```

Reply:
Makes a lot of sense! Wanna commit that?